central/signatureintegration/datastore/datastore_impl.go

```diff
@@ -84,6 +85,10 @@ func (d *datastoreImpl) AddSignatureIntegration(ctx context.Context, integration
 		return nil, errox.InvalidArgs.CausedBy(err)
 	}
 
+	if err := verifySignatureIntegrationOrigin(ctx, integration); err != nil {
+		return nil, errox.NotAuthorized.CausedByf("cannot create signature integration: %v", err)
```

Comment:
```suggestion
		return nil, errox.NotAuthorized.CausedByf("verifying integration origin: %v", err)
```

Note that in golang error convention, the `cannot create signature integration` part should be added to the wrapped error by the caller of `AddSignatureIntegration`.